templates/vue/src/components/Lightbox.vue

```diff
@@ -0,0 +1,249 @@
+
+      if (videoHeight > browserHeight * resizeRatio) {
+        resizeRatio *= browserHeight / videoHeight
+        videoWidth *= resizeRatio
+        videoHeight *= resizeRatio
+      }
+
+      const nodeSpace = NORMAL_RADIUS * 2 * 1.3
+      const adjustedVideoHeight = Math.min(videoHeight, browserHeight - nodeSpace)
+      const adjustedVideoWidth = Math.min(videoWidth, browserWidth - nodeSpace)
+
+      const heightAdjustmentRatio = adjustedVideoHeight / videoHeight
+      const widthAdjustmentRatio = adjustedVideoWidth / videoWidth
+      let adjustmentRatio = widthAdjustmentRatio
+      let adjustedOn = "width"
+
+      if (Helpers.getAspectRatio() < 1) {
+        adjustedOn = "height"
+        adjustmentRatio = heightAdjustmentRatio
+      }
+
+      adjustmentRatio *= 0.95
+      return {
+        adjustedOn,
+        width: videoWidth * adjustmentRatio,
+        height: videoHeight * adjustmentRatio,
+      }
+    },
+  },
+  async mounted() {
+    const node = await this.tapestryApiClient.getNode(this.nodeId)
+    const meta = await this.tapestryApiClient.getNodeProgress(this.nodeId)
+    node.typeData.progress[0].value = meta.progress
+    node.typeData.progress[1].value = 1.0 - meta.progress
+
+    const settings = await this.tapestryApiClient.getH5pSettings()
+    this.h5pSettings = settings
+
+    this.node = node
+    this.isLoaded = true
+    this.dimensions = {
+      ...this.dimensions,
+      left: (Helpers.getBrowserWidth() - this.lightboxDimensions.width) / 2,
+      width: this.lightboxDimensions.width,
+      height: this.lightboxDimensions.height,
+    }
+    thisTapestryTool.changeToViewMode(this.lightboxDimensions)
+  },
+  async beforeDestroy() {
+    await this.tapestryApiClient.updateUserProgress(
+      this.nodeId,
+      this.node && this.node.typeData.progress[0].value
+    )
+    thisTapestryTool.exitViewMode()
+  },
+  methods: {
+    updateH5pSettings(newSettings) {
+      this.h5pSettings = newSettings
+    },
+    updateDimensions({ width, height }) {
+      this.dimensions = {
+        ...this.dimensions,
+        width,
+        height,
+      }
+    },
+  },
+}
+</script>
+
+<style scoped>
+#lightbox {
+  position: fixed;
+  top: 0;
+  left: 0;
+  bottom: 0;
+  right: 0;
+  opacity: 1;
+  transform: translateY(0);
+}
+
+#spotlight-overlay {
+  width: 100%;
+  height: 100%;
+  position: fixed;
+}
+
+.content {
+  position: absolute;
+  top: 5vh;
+  left: 5vw;
+  height: 90vh;
+  width: 90vw;
+  z-index: 100;
+  background-position: 0 0;
+  background-size: cover;
+  background-color: black;
+  box-shadow: 0 0 100px -40px #000;
+  border-radius: 15px;
+}
+
+.media-wrapper {
+  background: #000;
+  outline: none;
+  border-radius: 15px;
+  overflow: hidden;
+}
+
+.media-wrapper-embed {
+  background: white;
+}
+
+.content-text {
+  outline: none;
+  background: #eee;
+  color: #333;
+  padding: 1em;
+}
+
+.content-text .media-wrapper {
+  height: 100%;
+  overflow: scroll;
+  background: transparent;
+}
```

Comment:
Couldn't this be moved into the TextMedia component?

Reply:
Unfortunately no because it deals with the styles in the outer lightbox container. I'll see if I can make any structural changes so that I can move this around.